lesson6/hometask_frameq.rb

```diff
@@ -0,0 +1,87 @@
+require 'YAML'
+
+class Person
+
+  attr_accessor :age, :sex, :fullname, :file, :name
+
+  def self.add(name, &block)
+    self.class.send(:attr_accessor, name)
+    instance_variable_set("@#{name.to_sym}", Person.new(name, &block))
+  end
+
+  def initialize(name, &block)
+    @name = name
+    instance_eval(&block)
+    file if @file.nil?
+    @info = {:name => @name, :fullname => @fullname, :age => @age, :sex => @sex, :file => @file}
+    save!
+  end
+
+  def sex(sex = nil)
```

Comment:
Read [this](https://github.com/nonkor/Betlab-ruby-courses/pull/27/files#r86520578). You have the same problem here. Would be better to handle this by using `UserFactory` class or some specific class which is created exactly for such need.